imap-proto/src/core.rs

```diff
@@ -87,6 +87,10 @@ named!(pub nstring<Option<&[u8]>>, alt!(
     map!(string, |s| Some(s))
 ));
 
+named!(pub string_utf8<&str>, map!(string, |s| str::from_utf8(s).unwrap()));
```

Comment:
It'd be nice if you introduce these helpers to convert existing `str::from_utf8()` users that would benefit from these. Ideally in that case the whole change would go in a separate commit.